### PR TITLE
Add collector for the logged_in_users table

### LINKED ISSUE #29578: Add collector for the logged_in_users table
|Wazuh version|Install type|Install method|Platform|
|---|---|---|---|
|4.14.0 | Agent| Packages/Sources | All |

# Description
Add support for retrieving system login session data via the logged_in_users table using the new os_data submodule in data_provider. This information will enrich the system inventory with real-time user session context, including which users are currently logged in, from where, and since when.

**Table schema:**
| Column         | Type     | Description                      |
|----------------|----------|----------------------------------|
| `type`         | text     | Login type                       |
| `user`         | text     | User login name                  |
| `tty`          | text     | Device name                      |
| `host`         | text     | Remote hostname                  |
| `time`         | bigint   | Time entry was made              |
| `pid`          | integer  | Process (or thread) ID           |
| `sid`          | text     | The user's unique security identifier |
| `registry_hive`| text     | HKEY_USERS registry hive         |

# Tasks

- [x] Implement the data extractors from Linux, Windows, and macOS.  
- [x] Implement logic in `extended_sources` to retrieve OS logged_in_users data.  
- [x] Add unit tests to validate field extraction and transformation.  

# Acceptance Criteria

- Results are structured as stateful in JSON.  
- Field names align with ECS where applicable.  
- Tests cover core logic and OS-specific edge cases.  
- Supported platforms: Linux, Windows, macOS.

Reference:
 - [Internal analysis](https://docs.google.com/spreadsheets/d/1XwgDInDeReqhZ6DoAn4SmqrKvCFmN40NjoUBWQiwt54/edit?gid=0#gid=0)
 - [Schema documentation](https://osquery.io/schema/5.16.0/#logged_in_users)
- [Spec definition](https://github.com/osquery/osquery/blob/master/specs/logged_in_users.table)

### PR DESCRIPTION
**Pull request by LucioDonda**
|Related issue|
|---|
|Closes #29578|

## Description

Add support for retrieving system login session data via the logged_in_users table using the new os_data submodule in data_provider. This information will enrich the system inventory with real-time user session context, including which users are currently logged in, from where, and since when.

- [x]  Implement the data extractors from Linux, Windows, and macOS.
- [x]  Implement logic in extended_sources to retrieve OS logged_in_users data.
- [x]  Add unit tests to validate field extraction and transformation.

## Example output

```json
    {
        "host": "",
        "pid": -1,
        "registry_hive": "HKEY_USERS\\S-1-5-21-2412136705-175749101-1827833236-500",
        "sid": "S-1-5-21-2412136705-175749101-1827833236-500",
        "time": 1747783964,
        "tty": "Console",
        "type": "active",
        "user": "Administrator"
    }
``` 

* From base approach some methods where updated in order to use 

| Newer SDK only | Older SDK |
| -- | -- |
| WTSEnumerateSessionsExW(...) | WTSEnumerateSessionsW(h, 0, 1, ...) |
| WTS_SESSION_INFO_1W | WTS_SESSION_INFOW | 
| WTSFreeMemoryEx(...) | WTSFreeMemory(...) |
| pSessionInfo[i].pSessionName | pSessionInfo[i].pWinStationName |

</body></html>

# Checks

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X

Review Checklist
  - [ ]  Code changes reviewed
  - [x]  Relevant evidence provided
  - [x]  Tests cover the new functionality
  - [ ]  Configuration changes documented
  - [ ]  Developer documentation reflects the changes
  - [x]  Meets requirements and/or definition of done
  - [x]  No unresolved dependencies with other issues

### CODE REVIEW

File: src/data_provider/src/extended_sources/users/tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+project(extended_sources_users_unit_test)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+include_directories(${CMAKE_SOURCE_DIR}/../external/googletest/googletest/include/)
+include_directories(${CMAKE_SOURCE_DIR}/../external/googletest/googlemock/include/)
+link_directories(${CMAKE_SOURCE_DIR}/../external/googletest/lib/)
+
+if(UNIX)
```
Comment (nbertoldo) [inline src/data_provider/src/extended_sources/users/tests/CMakeLists.txt:12]: We should unify the approach in all the `CMakeLists.txt`

**Review by nbertoldo — CHANGES_REQUESTED** [review]

File: src/data_provider/src/extended_sources/users/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+include_directories(include)
+include_directories(${CMAKE_SOURCE_DIR}/../external/nlohmann/)
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+  include_directories(${CMAKE_SOURCE_DIR}/../shared_modules/utils)
+endif()
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+  set(SRC_FILES src/logged_in_users_win.cpp)
+elseif(UNIX)
```
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/data_provider/src/extended_sources/users/CMakeLists.txt:17]: I couldn't find Unix as a System Name, I'll assume that if not WIndows then it should be darwin or linux (UNIX)

File: src/data_provider/src/extended_sources/users/tests/CMakeLists.txt (hunk 1 above)
Comment (LucioDonda) [reply src/data_provider/src/extended_sources/users/tests/CMakeLists.txt:12]: Same as above

File: src/data_provider/src/extended_sources/users/src/logged_in_users_win.cpp [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+      m_winSddlWrapper(std::move(winSddlWrapper)),
+      m_winSecurityWrapper(std::move(winSecurityWrapper)) {}
+
+LoggedInUsersProvider::LoggedInUsersProvider()
+    : m_twsApiWrapper(std::make_shared<TWSapiWrapper>()),
+      m_winBaseWrapper(std::make_shared<WinBaseApiWrapper>()),
+      m_winSddlWrapper(std::make_shared<WinSDDLWrapper>()),
+      m_winSecurityWrapper(std::make_shared<WinSecurityBaseApiWrapper>()) {}
+
+nlohmann::json LoggedInUsersProvider::collect()
```
Comment (cborla) [inline src/data_provider/src/extended_sources/users/src/logged_in_users_win.cpp:37]: perhaps the collect function could be separated to make it clearer, for example.

``` c++
std::vector<SessionInfo> LoggedInUsersProvider::enumerateSessions();
SessionInfoDetails LoggedInUsersProvider::buildSessionDetails(const SessionInfo& session);
nlohmann::json LoggedInUsersProvider::serializeSessionDetails(const SessionInfoDetails& details);

```

```c++
nlohmann::json LoggedInUsersProvider::collect() {
    nlohmann::json results;

    for (const auto& session : enumerateSessions()) {
        auto sessionDetails = buildSessionDetails(session);
        results.push_back(serializeSessionDetails(sessionDetails));
    }

    return results;
}

```

**Review by cborla — APPROVED** [review]
Review Checklist
  - [x]  Code changes reviewed
  - [x]  Relevant evidence provided
  - [x]  Tests cover the new functionality
  - [x]  Meets requirements and/or definition of done
  - [x]  No unresolved dependencies with other issues

**Review by nbertoldo — APPROVED** [review]
LGTM.

Review Checklist
  - [x]  Code changes reviewed
  - [x]  Relevant evidence provided
  - [x]  Tests cover the new functionality
  - [x]  Meets requirements and/or definition of done
  - [x]  No unresolved dependencies with other issues

**Review by Nicogp — APPROVED** [review]
Review Checklist
  - [x]  Code changes reviewed
  - [x]  Relevant evidence provided
  - [x]  Tests cover the new functionality
  - [x]  Meets requirements and/or definition of done
  - [x]  No unresolved dependencies with other issues
